### PR TITLE
Refactor: refine the fee sheet, new encounter form, procedure order pages.

### PR DESCRIPTION
**Pull request by stu01509**
Refine the fee sheet, new encounter form, procedure order pages.

Fixes #

#### Short description of what this resolves:
- Remove outdated code, comment, and inline-style.
- Adjust the bootstrap form class.

### Fee Sheet:
image

image

image

### New Encounter Form:
image

image

### Procedure Order:
image

image

image

image

### CODE REVIEW

**Comment by stu01509** [comment]
Hi @tywrenn 

Seems like are outdated jquery-ui style, can I remove it?https://github.com/openemr/openemr/blob/ef225c5ea5b771e84cafdd53000e52697eaebc00/interface/forms/fee_sheet/new.php#L572-L580 

Also, can you help me to check this line? This contains the `width="95%"`, when I remove it, the view will differ.
https://github.com/openemr/openemr/blob/ef225c5ea5b771e84cafdd53000e52697eaebc00/interface/forms/fee_sheet/new.php#L895

Before:
image

After:
image

File: interface/patient_file/encounter/find_code_history.php [hunk 1]
```diff
@@ -171,10 +171,10 @@ function setCodes() {
 </head>
 
 <body>
-    <div class="container-fluid w-100 position-fixed" style="margin-right: 10px;">
+    <div class="container-fluidposition-fixed">
         <div class="input-group bg-white">
             <div class="input-group-prepend">
-                <button class="btn btn-secondary" onclick='clearCodes(this)'><i class="fa fa-trash fa-1x"></i></button>
+                <button class="btn btn-danger" onclick='clearCodes(this)'><i class="fa fa-trash fa-1x"></i></button>
```
Comment (stu01509) [inline interface/patient_file/encounter/find_code_history.php:177]: Hi @bradymiller  and @tywrenn 

The #LINE186 contains `bg-warning` class, but it let the view looks like odd, should I remove it?

image

File: interface/forms/newpatient/common.php [hunk 2]
```diff
@@ -437,8 +434,7 @@ function cancelClickedOld() {
                         <?php
                         if (!$GLOBALS['gbl_visit_referral_source']) {
                             echo "style='display:none' ";
-                        } ?>
-                    >
+                        } ?>>
                         <div class="col-sm-2">
                             <label for="form_referral_source" class="oe-text-to-right"><?php echo xlt('Referral Source'); ?>:</label>
                         </div>
```
Comment (stu01509) [inline interface/forms/newpatient/common.php:440]: Hi @bradymiller  and @tywrenn 

Such as this `oe-text-to-right` class, Is it can convert to bootstrap `text-right` ?

File: interface/patient_file/encounter/find_code_history.php (hunk 1 above)
Comment (tywrenn) [reply interface/patient_file/encounter/find_code_history.php:172]: Seems useless so you can remove it

**Comment by tywrenn** [comment]
@stu01509 

> Seems like are outdated jquery-ui style, can I remove it?

Yep you can remove it

> Also, can you help me to check this line? This contains the width="95%", when I remove it, the view will differ.

I'll check it over and figure out how to convert it

File: interface/forms/newpatient/common.php (hunk 2 above)
Comment (tywrenn) [reply interface/forms/newpatient/common.php:440]: @stu01509 Yep the `oe-text-to-right` class is the same as `text-right`. Also the same for left. You can go ahead and replace those with BS4 classes instead.

Comment (tywrenn) [reply interface/forms/newpatient/common.php:440]: Not really sure about the other oe classes as they have some differences yet I know for sure these ones can be replaced.

File: interface/patient_file/encounter/find_code_history.php (hunk 1 above)
Comment (sjpadgett) [reply interface/patient_file/encounter/find_code_history.php:172]: @stu01509 
umm, that line doesn't do what you think! its for Help
image

Also, if you don't select a procedure then this dialog will highlight all history for user. Still the dialog would not be used unless user has already selected a procedure. This is how it looks with procedure selected. So, changing L186 has nothing to do with highlights. You can move the bg class up to L-182 and change color if wanted as long as it contrast with dialog so in just doesn't blend into dialog background.

image

so bottom line, please be careful making assumptions when removing anything and check JS to make sure something is not dynamically being used. Also, don't fool with the floating work bar. I want it to work the way it work now...

**Comment by stu01509** [comment]
Hi @sjpadgett  and @tywrenn 

I adjust a little bit js, I guess the diagnosis code procedure may like this? 
If user didn't select anything, the diagnosis code will not highlight any item, otherwise, it will highlight the user-selected item .

https://www.loom.com/share/9f6ceff8334d4376b79f17c280a895c6

**Comment by sjpadgett** [comment]
works for me thanks.

File: interface/forms/newpatient/common.php [hunk 3]
```diff
@@ -291,39 +289,38 @@ function cancelClickedOld() {
...
-                        <?php echo $encounter_followup ? (xlt("Follow up for") . ": " . $encounter_followup . " Dated: " . $followup_date) : ''; ?></small>
+                        <?php echo $encounter_followup ? (xlt("Follow up for") . ": " . $encounter_followup . " Dated: " . $followup_date) : ''; ?>
+                    </small>
                 </legend>
                 <div id="visit-details" class="px-5">
                     <div class="form-row align-items-center">
                         <div class="col-sm-2">
-                            <label for="pc_catid" class="oe-text-to-right"><?php echo xlt('Visit Category:'); ?></label>
+                            <label for="pc_catid" class="text-right"><?php echo xlt('Visit Category:'); ?></label>
                         </div>
```
Comment (stu01509) [inline interface/forms/newpatient/common.php:324]: Hi @bradymiller and @tywrenn 

Although I convert the `oe-text-to-right` class to the bootstrap `text-right` class, but I found the OpenEMR-interface document write detail about the `oe` class, Does the document outdated?

https://github.com/openemr/openemr/tree/master/interface#special-classes

Comment (tywrenn) [reply interface/forms/newpatient/common.php:324]: @stu01509 It _does do_ what the documentation says, _however_, this was used before BS4 was added. Now we can replace it with `text-left` and `text-right`.

**Comment by tywrenn** [comment]
@stu01509 Let me know when u want me to review this

**Comment by stu01509** [comment]
@tywrenn I already finished this PR, so go ahead :)

**Comment by tywrenn** [comment]
@stu01509 please rebase

**Comment by stu01509** [comment]
@tywrenn  Done.

File: interface/forms/procedure_order/new.php [hunk 4]
```diff
@@ -382,13 +382,13 @@ function addProcLine(flag = false) {
...
+    let cell = "<table class='table table-sm proc-table'><tr><td class='itemDelete'><button type='button' class='btn btn-danger btn-delete' onclick='deleteRow(event)'></button></td>" +
         "<td class='procedure-div'><input type='hidden' name='form_proc_order_title[" + i + "]' value='" + prc_name + "'>" +
         "<input type='text' class='form-control' name='form_proc_type_desc[" + i + "]' onclick='sel_proc_type(" + i + ")' " +
         "onfocus='this.blur()' title='<?php echo xla('Click to select the desired procedure'); ?>' style='cursor:pointer;' readonly /> " +
         "<input type='hidden' name='form_proc_type[" + i + "]' value='-1' /></td>" +
-        "<td class='diagnosis-div input-group'><span class='input-group-addon'>" +
-        "<i onclick='current_diagnoses(this)' class='fa fa-search fa-lg' title='<?php echo xla('Click to search past and current diagnoses history'); ?>'></i></span>" +
+        "<td class='diagnosis-div input-group'><span class='btn btn-secondary'>" +
+        "<i onclick='current_diagnoses(this)' class='input-group-prepend fa fa-search fa-lg' title='<?php echo xla('Click to search past and current diagnoses history'); ?>'></i></span>" +
```
Comment (tywrenn) [inline interface/forms/procedure_order/new.php:391]: This is not how input-group-prepend works.

File: interface/forms/procedure_order/new.php [hunk 5]
```diff
@@ -382,13 +382,13 @@ function addProcLine(flag = false) {
     let i = 0;
     for (; f['form_proc_type[' + i + ']']; ++i) ;
     // build new item html.. a hidden html block to clone may be better here.
-    let cell = "<table class='table table-sm proc-table'><tr><td onclick='deleteRow(event)' class='itemDelete'><i class='fa fa-remove'></i></td>" +
+    let cell = "<table class='table table-sm proc-table'><tr><td class='itemDelete'><button type='button' class='btn btn-danger btn-delete' onclick='deleteRow(event)'></button></td>" +
```
Comment (tywrenn) [inline interface/forms/procedure_order/new.php:385]: Add table-responsive to the table

File: interface/forms/procedure_order/new.php [hunk 6]
```diff
@@ -536,110 +536,120 @@ function validate(f) {
...
-            </div>
+            <?php echo  $oemr_ui->pageHeading() . "\r\n"; ?>
         </div>
     </div>
     <div class="row">
         <div class="col-sm-12">
             <form class="form-horizontal" method="post" action="<?php echo $rootdir ?>/forms/procedure_order/new.php?id=<?php echo attr_url($formid); ?>" onsubmit="return validate(this)">
                 <input type="hidden" name="csrf_token_form" value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
-                <fieldset>
+                <div class="container">
```
Comment (tywrenn) [inline interface/forms/procedure_order/new.php:550]: This doesn't actually work. You have to have the fieldset tag to allow the legend tag to work properly. So the first tag that should be here is the fieldset tag followed by the div container.

File: interface/patient_file/encounter/find_code_history.php [hunk 7]
```diff
@@ -79,10 +79,6 @@ function get_history_codes($pid)
     <meta charset="utf-8" />
     <?php Header::setupHeader(['opener']); ?>
     <style>
-        body {
-            height: 100%;
-            width: 100%;
-        }
         .tips {
```
Comment (tywrenn) [inline interface/patient_file/encounter/find_code_history.php:82]: Try to replace this `tips` class with the BS4 `d-none` class in both the html and scripts.

Comment (tywrenn) [reply interface/patient_file/encounter/find_code_history.php:82]: Be careful replacing it though cause there's an id called tips as well so you don't wanna replace that.

File: interface/patient_file/encounter/find_code_history.php [hunk 8]
```diff
@@ -202,15 +198,15 @@ function setCodes() {
         <div class="loading text-center"><i class="fa fa-sync fa-3x fa-spin"></i></div>
```
Comment (tywrenn) [inline interface/patient_file/encounter/find_code_history.php:198]: Replace font awesome spinner with BS4 spinner

File: interface/forms/fee_sheet/new.php [hunk 9]
```diff
@@ -920,7 +904,6 @@ function pricelevel_changed(sel) {
                                     $cleave_opt = is_numeric(substr($fs_option, 0, 2)) ? 2 : 1;
                                     echo ($i <= 1) ? " <tr>\n" : "";
                                     echo "  <td width='50%'  nowrap>\n";
```
[the remark this answers is not in the document]
Comment (stu01509) [reply interface/forms/fee_sheet/new.php:906]: @tywrenn  
When I change the line 906, change the width 50% to w-50, the view will be changing. They are looks different, need you to confirm.

Before:
image

After:
image

**Comment by stu01509** [comment]
Update the commit.

File: interface/forms/newpatient/common.php [hunk 10]
```diff
@@ -146,9 +146,6 @@ function refreshIssue(issue, title) {
                 e.preventDefault();
                 e.stopPropagation();
                 dlgopen('', '', 700, 650, '', '', {
-
-                    buttons: [{text: <?php echo xlj('Close'); ?>, close: true, style: 'default btn-sm'}],
```
Comment (tywrenn) [inline interface/forms/newpatient/common.php:150]: @stu01509 Dunno if you intended to delete this

File: interface/forms/procedure_order/new.php [hunk 11]
```diff
@@ -382,16 +382,33 @@ function addProcLine(flag = false) {
     let i = 0;
```
Comment (tywrenn) [inline interface/forms/procedure_order/new.php:382]: Padding is still wrong in this file
image

File: interface/patient_file/encounter/find_code_history.php [hunk 12]
```diff
@@ -78,20 +78,6 @@ function get_history_codes($pid)
     <title><?php xlt("Find Code History"); ?></title>
```
Comment (tywrenn) [inline interface/patient_file/encounter/find_code_history.php:78]: Something bad happened here @bradymiller @stu01509 

image

Comment (stu01509) [reply interface/patient_file/encounter/find_code_history.php:78]: In my environment running normally, need @bradymiller  to confirm

Comment (tywrenn) [reply interface/patient_file/encounter/find_code_history.php:78]: @bradymiller

Comment (bradymiller) [reply interface/patient_file/encounter/find_code_history.php:78]: Hi,
I'd be willing to bet that @tywrenn is seeing this since his instance has not imported the ICD10 codes (Administration->Coding->External Data Loads->ICD10->Install).
And that @stu01509 is not seeing an error since has imported the ICD10 codes
If this is the case, then is a bug.

Comment (bradymiller) [reply interface/patient_file/encounter/find_code_history.php:78]: And if is a bug, likely pre-existing. I'll test it out later.

Comment (bradymiller) [reply interface/patient_file/encounter/find_code_history.php:78]: Confirmed behavior reported by @tywrenn when clicking the following button below Diagnosis Code :

**Comment by stu01509** [comment]
Update the commit.

File: interface/forms/fee_sheet/new.php [hunk 13]
```diff
@@ -1472,44 +1456,54 @@ function pricelevel_changed(sel) {
...
-                                            <?php echo xlt('Re-Open Visit'); ?></button>
+                                            <?php echo xlt('Re-Open Visit'); ?>
+                                        </button>
                                     <?php } // end billed without charges ?>
                                     <button type='submit' class='btn btn-secondary btn-add' name='bn_addmore' onclick='return this.clicked = true;' value='<?php echo xla('Add More Items'); ?>'>
-                                        <?php echo xlt('Add More Items'); ?></button>
+                                        <?php echo xlt('Add More Items'); ?>
+                                    </button>
                                 <?php } // end billed ?>
                                     <button type='button' class='btn btn-link btn-cancel' onclick="top.restoreSession();location='<?php echo $GLOBALS['form_exit_url']; ?>'">
```
Comment (tywrenn) [inline interface/forms/fee_sheet/new.php:1498]: Should have btn-secondary here instead of btn-link

File: interface/forms/procedure_order/new.php (hunk 11 above)
Comment (tywrenn) [reply interface/forms/procedure_order/new.php:382]: @stu01509 Still wrong

**Review by tywrenn — CHANGES_REQUESTED** [review]

**Comment by stu01509** [comment]
Hi @tywrenn 

Not sure about that, but in my environment, it seems like good,
Any step I wrong?
https://www.loom.com/share/9e7a7ea5c7e543b5b8db2321d89143fa

**Comment by tywrenn** [comment]
@stu01509 This is for the screen behind the popup, not the popup.

**Comment by tywrenn** [comment]
The padding is off there

**Comment by stu01509** [comment]
Hey @tywrenn  Do you mean the modal the content didn't correct?

**Comment by tywrenn** [comment]
@stu01509 
https://www.loom.com/share/060da95719a047f2a1f558f2c740813c

**Comment by stu01509** [comment]
Hi @tywrenn 

Update the commit.
I am not sure is it the best way to add the padding, but I add the `p-2` in each `form-row ` class.

image

**Review by tywrenn — APPROVED** [review]
I'll fix some of the residual issues in a separate pull. Approving this @bradymiller

File: interface/forms/fee_sheet/new.php [hunk 14]
```diff
@@ -1472,44 +1456,54 @@ function pricelevel_changed(sel) {
...
+                                    <button type='button' class='btn btn-secondary btn-cancel' onclick="top.restoreSession();location='<?php echo $GLOBALS['form_exit_url']; ?>'">
                                     <?php echo xlt('Cancel');?></button>
                                     <input type='hidden' name='form_has_charges' value='<?php echo $fs->hasCharges ? 1 : 0; ?>' />
                                     <input type='hidden' name='form_checksum' value='<?php echo attr($current_checksum); ?>' />
                                     <input type='hidden' name='form_alertmsg' value='<?php echo attr($alertmsg); ?>' />
                             </div>
+                            <input type='hidden' name='form_has_charges' value='<?php echo $fs->hasCharges ? 1 : 0; ?>' />
+                            <input type='hidden' name='form_checksum' value='<?php echo attr($current_checksum); ?>' />
+                            <input type='hidden' name='form_alertmsg' value='<?php echo attr($alertmsg); ?>' />
```
Comment (bradymiller) [inline interface/forms/fee_sheet/new.php:1506]: The 3 above lines are duplicated directly from above

File: interface/forms/newpatient/common.php (hunk 10 above)
Comment (bradymiller) [reply interface/forms/newpatient/common.php:150]: Was this meant to be deleted?

Comment (bradymiller) [reply interface/forms/newpatient/common.php:150]: this looks good on testing. see why this Close was removed now.

Comment (stu01509) [reply interface/forms/newpatient/common.php:150]: Hi @bradymiller  and @tywrenn 

Ok, I undo this change.
if I remember correctly that modal has two close buttons that are the reason why I remove it.

**Comment by bradymiller** [comment]
@stu01509 and @tywrenn , Really nice stuff! Gonna bring this in and will follow it with a commit or 2 to fix up some minor issues. thanks for the improvements!
